### PR TITLE
[57] solve type issue coming up at docker build time

### PR DESCRIPTION
**Pull request by caravinci**
When setting the new 'arranger manifest columns' environment variable, I missed adding it to the output types object; and that oversight is breaking the deployment process.

Additionally, in order to respect the current "all vars are strings" pattern, I've relocated the logic that processes the var into the table component, where the value will be used.

### CODE REVIEW

File: global/config.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export const getConfig = () => {
     NEXT_PUBLIC_ARRANGER_INDEX: string;
     NEXT_PUBLIC_ARRANGER_API: string;
     NEXT_PUBLIC_ARRANGER_ADMIN_UI: string;
+    NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS: string;
```
Comment (caravinci) [inline global/config.ts:32]: This is the pesky little type setting I was missing. 😅

File: next.config.js [hunk 2]
```diff
@@ -11,8 +10,7 @@ module.exports = withCSS({
     NEXT_PUBLIC_ARRANGER_INDEX: process.env.NEXT_PUBLIC_ARRANGER_INDEX,
     NEXT_PUBLIC_ARRANGER_API: process.env.NEXT_PUBLIC_ARRANGER_API_URL,
     NEXT_PUBLIC_ARRANGER_ADMIN_UI: process.env.NEXT_PUBLIC_ARRANGER_ADMIN_UI_URL,
-    NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS: env.get('NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS').default('').asArray()
-        .map(fieldName => fieldName.trim().replace(/['"]+/g, '')),
+    NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS: process.env.NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS || '',
```
Comment (caravinci) [inline next.config.js:13]: which means this one needs to come out as a string, not as an already sanitised array...

File: components/pages/explorer/RepoTable.tsx [hunk 3]
```diff
@@ -208,10 +208,13 @@ const getTableStyle = (theme: typeof defaultTheme) => css`
 
 const RepoTable = (props: PageContentProps) => {
   const { NEXT_PUBLIC_ARRANGER_API, NEXT_PUBLIC_ARRANGER_PROJECT_ID, NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS } = getConfig();
+  const manifestColumns = NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS
+    .split(',')
+    .map(fieldName => fieldName.trim().replace(/['"]+/g, ''));
```
Comment (caravinci) [inline components/pages/explorer/RepoTable.tsx:213]: and then this logic should parse the string into an array, and remove the noise.

**Review by caravinci — COMMENTED** [review]
Comments for great good. 💰

File: components/pages/explorer/RepoTable.tsx [hunk 4]
```diff
@@ -208,10 +208,13 @@ const getTableStyle = (theme: typeof defaultTheme) => css`
 
 const RepoTable = (props: PageContentProps) => {
   const { NEXT_PUBLIC_ARRANGER_API, NEXT_PUBLIC_ARRANGER_PROJECT_ID, NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS } = getConfig();
+  const manifestColumns = NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS
+    .split(',').filter(field => field.trim()) // break it into arrays, and ensure there's not empties
+    .map(fieldName => fieldName.replace(/['"]+/g, '').trim());
```
Comment (caravinci) [inline components/pages/explorer/RepoTable.tsx:213]: and then this logic should parse the string into an array, and remove the noise.

**Review by anncatton — APPROVED** [review]

**Pull request merged**